src/EnergyPlus/ConvectionCoefficients.cc

```diff
@@ -769,35 +759,32 @@ void GetUserConvCoeffs(EnergyPlusData &state)
                                                                  ipsc->lAlphaFieldBlanks,
                                                                  ipsc->cAlphaFieldNames,
                                                                  ipsc->cNumericFieldNames);
-        auto &inUserCurve = state.dataConvectionCoefficient->hcIntUserCurve(Loop);
+        auto &inUserCurve = state.dataConvect->hcIntUserCurve(Loop);
         inUserCurve.Name = ipsc->cAlphaArgs(1);
+
+        ErrorObjectHeader eoh{RoutineName, CurrentModuleObject, inUserCurve.Name};
         inUserCurve.refTempType = static_cast<RefTemp>(getEnumerationValue(RefTempNamesUC, ipsc->cAlphaArgs(2)));
         if (inUserCurve.refTempType == RefTemp::Invalid) {
             ShowSevereError(state,
                             format("GetUserSuppliedConvectionCoefficients: {}: Invalid Key choice Entered, for {}={}",
                                    CurrentModuleObject,
                                    ipsc->cAlphaFieldNames(2),
                                    ipsc->cAlphaArgs(2)));
+
             ErrorsFound = true;
         }
 
         if (!ipsc->lAlphaFieldBlanks(3)) {
             inUserCurve.hcFnTempDiffCurveNum = Curve::GetCurveIndex(state, ipsc->cAlphaArgs(3));
             if (inUserCurve.hcFnTempDiffCurveNum == 0) {
-                ShowSevereError(state,
-                                format("GetUserSuppliedConvectionCoefficients: {}: Invalid Name Entered, for {}={}",
-                                       CurrentModuleObject,
-                                       ipsc->cAlphaFieldNames(3),
-                                       ipsc->cAlphaArgs(3)));
+                ShowSevereItemNotFound(state, eoh, ipsc->cAlphaFieldNames(3), ipsc->cAlphaArgs(3));
```

Comment:
Here is an example of the new error pattern. `eoh` is the ErrorObjectHeader struct that contains `routineName`, `objectType` (i.e., `CurrentModuleObject`), and `objectName` fields.